### PR TITLE
Fixes #9839: Build slapd with lmdb

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9839

### CODE REVIEW

File: rudder-inventory-ldap/SOURCES/rudder-inventory-ldap.init [hunk 1]
```diff
@@ -382,25 +382,24 @@ start_slapd() {
...
+      # calculate the value to match about 50% of RAM (this is only virtual memory)
       MEMSIZE=$(sed -n '/^MemTotal/s/[^0-9]//gp' /proc/meminfo)
-      # Round down using integer division (800 bytes is the size of an entry)
-      CACHESIZE=$((${MEMSIZE} * 1024 / 800 / 10 / 10000 * 10000))
-      # Maximum cache size of 40M entries. Above about 40G of RAM (about 53.6M entries) OpenLDAP dumps core, so we force an upper limit.
-      if [ ${CACHESIZE} -gt 40000000 ]; then
-        CACHESIZE=40000000
+      MDBSIZE=$((${MEMSIZE} * 1024 / 2 ))
+      # on 32 bits architectures max it to 500M (PAE allows more ram but not more virtual memory)
+      if [ "$(getconf LONG_BITS)" -ne "64" ] && [ "${MDBSIZE}" -gt 524288000 ]
```
Comment (ncharles) [inline rudder-inventory-ldap/SOURCES/rudder-inventory-ldap.init]: this is LONG_BIT

**Comment by peckpeck** [comment]
Commit modified

**Comment by amousset** [comment]
Don't we need a migration script for `slapd.conf`?

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**